### PR TITLE
Use extended metadata for playwright/test

### PR DESCRIPTION
**Pull request by ryanjduffy**
Adopts the changes from #24 to add the proper extended metadata for tests for recordings.

* Use `@replayio/replay/metadata` to create the test metadata structure
* Move `testId` into an `x-playwright` custom block to avoid the CLI complaining about invalid custom metadata fields
* Updates the `test` metadata structure to use `run` as an object with an `id` that is a unique UUID and `title` that is a optionally user-defined string.
* Adds support for passing additional metadata (e.g. the git source for the test execution) via the existing `RECORD_REPLAY_METADATA` env variable or the new `PLAYWRIGHT_REPLAY_METADATA` or via `playwright.config.ts` config
* Adds support for configuring the run title either via `PLAYWRIGHT_REPLAY_RUN_TITLE` or via the config

### Adds some git metadata to every recording
```
PLAYWRIGHT_REPLAY_METADATA='{"git": {"repo": "https://github.com/RecordReplay/replay-cli"}}' npx playwright test --project replay-firefox --reporter @replayio/playwright/reporter,line
```
Using our "standard" env variable
```
RECORD_REPLAY_METADATA='{"git": {"repo": "https://github.com/RecordReplay/replay-cli"}}' npx playwright test --project replay-firefox --reporter @replayio/playwright/reporter,line
```
Using both variables. This'll warn and only use `PLAYWRIGHT_REPLAY_METADATA` as the most specific field
```
PLAYWRIGHT_REPLAY_METADATA='{"git": {"repo": "https://github.com/RecordReplay/replay-cli"}}' RECORD_REPLAY_METADATA='{"not-used": {"repo": "https://github.com/RecordReplay/replay-cli"}}' npx playwright test --project replay-firefox --reporter @replayio/playwright/reporter,line
```

Using the config

```ts
// playwright.config.ts
import { PlaywrightTestConfig } from "@playwright/test";
import { devices as replayDevices } from "@replayio/playwright";

const config: PlaywrightTestConfig = {
  reporter: [
    ["line"],
    ["@replayio/playwright/reporter", { runTitle: "hey", metadata: { git: { repo: "okay" } } }],
  ],
 // ... etc ...
}
```

### Set the run title

Using env variable
```
PLAYWRIGHT_REPLAY_RUN_TITLE="my custom run" npx playwright test --project replay-firefox --reporter @replayio/playwright/reporter,line
```

Using the config

```ts
// playwright.config.ts
import { PlaywrightTestConfig } from "@playwright/test";
import { devices as replayDevices } from "@replayio/playwright";

const config: PlaywrightTestConfig = {
  reporter: [
    ["line"],
    ["@replayio/playwright/reporter", { runTitle: "hey" }],
  ],
 // ... etc ...
}
```

### CODE REVIEW

File: packages/playwright/types/replayio__replay__metadata.d.ts
Comment (jcmorrow) [inline packages/playwright/types/replayio__replay__metadata.d.ts:20]: If running from GH actions probably the `github.run_id` would be best https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

File: packages/playwright/src/reporter.ts
Comment (jcmorrow) [inline packages/playwright/src/reporter.ts:143]: what is the shape of `titlePath()` here usually? Is it possible that we'll blow up trying to pull [2] from `undefined` or anything like that?

File: packages/playwright/src/reporter.ts
Comment (jcmorrow) [inline packages/playwright/src/reporter.ts:113]: What would this indicate?

**Review by jcmorrow — APPROVED** [review]
I'm not going to pretend to understand what every line of this does, but I called out the things that seemed slightly 🐟 y to me. I leave it to your discretion as to what should actually block merge. Also - your PR body is great in terms of examples. How would you feel about also adding that somewhere in a README (not sure if it should be in this repo or some related repo).

File: packages/playwright/src/reporter.ts
Comment (ryanjduffy) [reply packages/playwright/src/reporter.ts:113]: This reporter would still be called for a skipped test but there wouldn't be a recording so there's no work to do so we're bailing early.

Comment (ryanjduffy) [reply packages/playwright/src/reporter.ts:113]: Thinking more though, we do support the `timedOut` status so this should be `if (status === "skipped") return` which  is more clear anyway.

**Comment by ryanjduffy** [comment]
I'm holding off on documenting all of this stuff in the readme and the docs until it's out the door and I've had a chance to try it out with a couple test projects. I did document them here as a shortcut to future doc updates :)

File: packages/playwright/src/reporter.ts
Comment (jcmorrow) [reply packages/playwright/src/reporter.ts]: that is much clearer 🏅

**Pull request merged**
